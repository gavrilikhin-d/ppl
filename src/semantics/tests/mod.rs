use crate::test_compilation_result;

test_compilation_result!(candidate_not_viable);
test_compilation_result!(constraints);
test_compilation_result!(constraints_in_constructor);
test_compilation_result!(generics);
test_compilation_result!(missing_fields);
test_compilation_result!(multiple_errors);
test_compilation_result!(multiple_initialization);
test_compilation_result!(non_class_constructor);
test_compilation_result!(predeclare_vars);
test_compilation_result!(references);
test_compilation_result!(traits);
test_compilation_result!(type_as_value);
test_compilation_result!(wrong_initializer_type);
