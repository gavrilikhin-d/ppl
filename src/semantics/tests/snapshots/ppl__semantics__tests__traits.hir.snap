---
source: src/semantics/tests/mod.rs
expression: hir
---
trait Test:
	fn required <:Self> -> None

	fn default <x: Self> -> Self:
		return (x:Self)



fn required <x: Integer> -> None:
	return none

let res: Integer = default 1
println clone (res:Integer)
